ext/pdo_firebird/firebird_statement.c

```diff
@@ -608,6 +608,60 @@ static int firebird_stmt_param_hook(pdo_stmt_t *stmt, struct pdo_bound_param_dat
 				}
 			}
 
+#ifdef SQL_BOOLEAN
+			/* keep native BOOLEAN type */
+			if ((var->sqltype & ~1) == SQL_BOOLEAN) {
+				switch (Z_TYPE_P(parameter)) {
+					case IS_LONG:
+					case IS_DOUBLE:
+					case IS_TRUE:
+					case IS_FALSE:
+						*(FB_BOOLEAN*)var->sqldata = zend_is_true(parameter) ? FB_TRUE : FB_FALSE;
+						break;
+					case IS_STRING:
+						if ((Z_STRLEN_P(parameter) == 0)) {
+							*(FB_BOOLEAN*)var->sqldata = FB_FALSE;
+							break;
+						}
+
+						long lval;
+						double dval;
```

Comment:
I think we still avoid mixing declarations and statements, so please move these two declarations to the beginning of a block.